lib/fetch-package-metadata.js

```diff
@@ -118,7 +118,15 @@ function fetchNamedPackageData (dep, next) {
     }
     function pickVersionFromRegistryDocument (pkg) {
       if (!regCache[url]) regCache[url] = pkg
-      var versions = Object.keys(pkg.versions).sort(semver.rcompare)
+      var versions = Object.keys(pkg.versions)
+
+      var invalidVersions = versions.filter(function (v) { return !semver.valid(v) })
+      if (invalidVersions.length > 0) {
+        log.warn('', 'The package %s has invalid semver-version(s): %s. This usually only happens for unofficial private registries. ' +
```

Comment:
There should be an initial argument to this– something like `pickVersion` would be fine, I think.